### PR TITLE
Api changes

### PR DESCRIPTION
**Pull request by nadeemkhedr**
- Add server side validation
- Add timestamp fields for opportunities (used in dashboard)
- All data returned are json api compliant

### CODE REVIEW

File: models/opportunity.js [hunk 1]
```diff
@@ -2,13 +2,29 @@
...
 
 const OpportunitySchema = new Schema({
-    dateFrom: Date,
+    name: {
+        type: String,
+        required: 'Name is required'
+    },
+    dateFrom: {
+        type: Date,
+        validate: [dateValidator, 'Date to must be greater than Date from']
```
Comment (kamilogorek) [inline models/opportunity.js]: There's great module from `hapijs` creators called `joy` for performing object validations as well – https://github.com/hapijs/joi
However feel free to use mongoose validate functions if you prefer so :)

File: models/opportunity.js [hunk 2]
```diff
@@ -2,13 +2,29 @@
...
     dateTo: Date,
-    name: String,
-    skills: [SkillSchema]
+    skills: Array
 });
 
+//add createdAt, updatedAt
+OpportunitySchema.plugin(timestamps);
+
+function dateValidator(value) {
```
Comment (kamilogorek) [inline models/opportunity.js]: I'd use more descriptive name, as `dateValidator` is very vague. Something like `earlierThanDateTo` would work much nicer imo.

File: routes/developers.js [hunk 3]
```diff
@@ -5,31 +5,31 @@ const developersController = require('../controllers/developers');
 module.exports = (server) => {
```
Comment (kamilogorek) [inline routes/developers.js]: Those routes should stay as they were, but it should be registered as a plugin inside `index` instead and using `prefix` attribute.

See: http://hapijs.com/tutorials/plugins#plugin-options

Comment (nadeemkhedr) [inline routes/developers.js]: done!

**Comment by nadeemkhedr** [comment]
@kamilogorek the reason I added empty callback because its required, I changed that to `console.error` the error message, with a todo comment to log that when we have a logging system

Note: I am not writing any handling code, because the error should not happen, and honestly don't know how to handle it, because we are using internal hapi prefex as a plugin, so most likely if that error occurs it would be dependencies issue not user request issue

**Comment by kamilogorek** [comment]
It is? Then they have an error in their docs :)

Merging

**Pull request merged**